File: 0053-maximum-subarray/solution.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        
        int res = -INT_MAX , x = 0 , xx = 0 ; 
        for( auto i : nums ){
            x = max( i , x  + i);
            res = max( res , x);
        }
        return res ; 
        
    }
};

File: 0053-maximum-subarray/solution.java
class Solution {

    public int maxSubArray(int[] nums) {
        int n = nums.length , sum = 0, res = Integer.MIN_VALUE;
        for(int i = 0 ; i < n ; i++){
           sum = Math.max(nums[i],nums[i]+sum);
           res = Math.max(res,sum);
        }
        return res;
    }

}
